src/Http/Http.Extensions/src/RequestDelegateFactoryOptions.cs

```diff
@@ -36,4 +36,14 @@ public sealed class RequestDelegateFactoryOptions
     /// The list of filters that must run in the pipeline for a given route handler.
     /// </summary>
     public IReadOnlyList<Func<RouteHandlerContext, RouteHandlerFilterDelegate, RouteHandlerFilterDelegate>>? RouteHandlerFilterFactories { get; init; }
+
+    /// <summary>
+    /// The default endpoint metadata to add as part of the creation of the <see cref="RequestDelegateResult.RequestDelegate"/>.
+    /// </summary>
+    /// <remarks>
+    /// This metadata will be included in <see cref="RequestDelegateResult.EndpointMetadata" /> after any metadata inferred during creation of the
+    /// <see cref="RequestDelegateResult.RequestDelegate"/> but before any metadata provided by types in the delegate signature that implement
+    /// <see cref="IEndpointMetadataProvider" /> or <see cref="IEndpointParameterMetadataProvider" />.
+    /// </remarks>
+    public IReadOnlyList<object>? DefaultEndpointMetadata { get; init; }
```

Comment:
What about?

```suggestion
    public IEnumerable<object>? InitialEndpointMetadata { get; init; }
```

I don't see how it's a default given that it needs to be supplied every call to RDF.Create() to be used. I also like `IEnumerable` since that's all we need and it's more likely to be provided by the developer than read.